*Problem statement* :
<br>The following dataset consists of 1030 instances with 9 attributes and has no missing values. There are 8 input variables and 1 output variable. Seven input variables represent the amount of raw material (measured in kg/m³) and one represents Age (in Days). The target variable is Concrete Compressive Strength measured in (Mega Pascal). Build a Neural network model to predict the compressive strength.

In [1]:
#importing required libraries
import numpy as np
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense
from tensorflow.keras.layers import ReLU
from tensorflow.keras.layers import Dropout
from sklearn.model_selection import train_test_split
import pandas as pd

In [2]:
#loading dataset
data = pd.read_csv("D:\\DataScience\\Class\\assignment working\\ANN\\concrete.csv")
data.head()

cement   slag    ash  water  superplastic  coarseagg  fineagg  age  \
0   141.3  212.0    0.0  203.5           0.0      971.8    748.5   28   
1   168.9   42.2  124.3  158.3          10.8     1080.8    796.2   14   
2   250.0    0.0   95.7  187.4           5.5      956.9    861.2   28   
3   266.0  114.0    0.0  228.0           0.0      932.0    670.0   28   
4   154.8  183.4    0.0  193.3           9.1     1047.4    696.7   28   

   strength  
0     29.89  
1     23.51  
2     29.22  
3     45.85  
4     18.29

In [3]:
#seperating target and input data
x=data.drop("strength",axis=1)
y=data.strength

In [4]:
#splitting data into train and test
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test = train_test_split(x,y,test_size=0.3,random_state =1)
x_train.shape

(721, 8)

In [5]:
#standardizing data
from sklearn.preprocessing import StandardScaler
scale=StandardScaler()
x_train=scale.fit_transform(x_train)
x_test =  scale.fit_transform(x_test)

In [15]:
#defining model
def design_mlp():
    # Initializing the model 
    model = Sequential()
    model.add(Dense(15,input_dim =8,activation="relu"))
    model.add(Dense(50,activation="relu"))
    model.add(Dense(7,activation="relu"))
    model.add(Dense(1,activation="relu"))
    model.compile(loss="mean_squared_error",optimizer="RMSprop",metrics=["RootMeanSquaredError"])
    return model

model = design_mlp()

In [16]:
model.summary()

Model: "sequential_1"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
dense_3 (Dense)              (None, 15)                135       
_________________________________________________________________
dense_4 (Dense)              (None, 50)                800       
_________________________________________________________________
dense_5 (Dense)              (None, 7)                 357       
_________________________________________________________________
dense_6 (Dense)              (None, 1)                 8         
Total params: 1,300
Trainable params: 1,300
Non-trainable params: 0
_________________________________________________________________


In [17]:
#fitting model
model.fit(x=x_train,y=y_train,batch_size=1,epochs=100)


Epoch 1/100
721/721 [==============================] - 1s 2ms/step - loss: 634.0531 - root_mean_squared_error: 25.1804
Epoch 2/100
721/721 [==============================] - 1s 2ms/step - loss: 207.6005 - root_mean_squared_error: 14.4083
Epoch 3/100
721/721 [==============================] - 1s 2ms/step - loss: 180.0058 - root_mean_squared_error: 13.4166
Epoch 4/100
721/721 [==============================] - 1s 1ms/step - loss: 175.5081 - root_mean_squared_error: 13.2479
Epoch 5/100
721/721 [==============================] - 1s 1ms/step - loss: 158.0456 - root_mean_squared_error: 12.5716
Epoch 6/100
721/721 [==============================] - 1s 1ms/step - loss: 147.3062 - root_mean_squared_error: 12.1370
Epoch 7/100
721/721 [==============================] - 1s 2ms/step - loss: 145.2806 - root_mean_squared_error: 12.0532
Epoch 8/100
721/721 [==============================] - 1s 2ms/step - loss: 139.8295 - root_mean_squared_error: 11.8250
Epoch 9/100
721/721 [===========================

In [32]:
model.evaluate(x_test,y_test)

10/10 [==============================] - 0s 803us/step - loss: 36.7381 - root_mean_squared_error: 6.0612


[36.738067626953125, 6.061193466186523]

we can see our model is giving 6.06 as error value